src/engine.cpp

```diff
@@ -655,7 +679,20 @@ void StartupOneEngine(Engine *e, Date aging_date)
 		e->flags |= ENGINE_AVAILABLE;
 	}
 
-	RestoreRandomSeeds(saved_seeds);
+	/* Get parent variant index for syncing reliability via random seed. */
+	const Engine *re = e;
+	while (re->info.variant_id != INVALID_ENGINE && re->info.variant_id != re->index && (re->info.extra_flags & ExtraEngineFlags::SyncReliability) != ExtraEngineFlags::None) {
+		re = Engine::Get(re->info.variant_id);
+	}
+
+	SetRandomSeed(_settings_game.game_creation.generation_seed ^ seed ^
+	              (re->index << 16) ^
+				  (re->info.base_intro << 12) ^
+				  (re->info.decay_speed << 8) ^
+				  (re->info.lifelength << 4) ^
+				  re->info.retire_early ^
+	              e->type ^
+	              e->GetGRFID());
```

Comment:
some very strange line breaks and indenting here

Reply:
maybe put the operators at begin of line